conda/recipes/rmm/recipe.yaml

```diff
@@ -20,8 +20,23 @@ source:
 build:
   string: cuda${{ cuda_major }}_py${{ py_buildstring }}_${{ date_string }}_${{ head_rev }}
   script:
-    content:
-      - ./build.sh -v clean rmm --cmake-args=\"-DCMAKE_INSTALL_LIBDIR=lib\"
+    content: |
+      # Remove `-fdebug-prefix-map` line from CFLAGS and CXXFLAGS so the
+      # incrementing version number in the compile line doesn't break the
+      # cache
+      set -x
+      export CFLAGS=$(echo $CFLAGS | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+      export CXXFLAGS=$(echo $CXXFLAGS | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+      set +x
+
+      # Use relative paths for `__FILE__` by passing `-fmacro-prefix-map`
+      # https://gcc.gnu.org/onlinedocs/gcc-13.3.0/gcc/Preprocessor-Options.html#index-fmacro-prefix-map
+      set -x
+      export CFLAGS="${CFLAGS} -fmacro-prefix-map=$(pwd)/=''"
+      export CXXFLAGS="${CXXFLAGS} -fmacro-prefix-map=$(pwd)/=''"
+      set +x
```

Comment:
As `$(pwd)` is the top-level directory of this repo, passing the `-fmacro-prefix-map` option simply removes the `$(pwd)` portion of the path in all `__FILE__` expansions. Thus leaving us with paths relative to the top of the repo

A couple things we may want to consider:

1. Moving this into our CMake logic for all builds
2. Applying this behavior to more prefix mappings. [`-ffile-prefix-map`]( https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Overall-Options.html#index-ffile-prefix-map ) would do this for all prefix-mapping cases GCC handles

Beyond fixing the `__FILE__` expansion this gets us closer to reproducible builds. Also it should behave more robustly with caching (though we should test to confirm)

Reply:
I would consider merging this as-is, and doing (1) or (2) as follow-up work.

Reply:
I would also support this approach here. I think this is a good idea. I suggest that we try and put this into rapids-cmake so that we get this for free in all our libraries. The main challenge I see with that is identifying the appropriate root path to use. I agree with Bradley that I'm OK with doing that as a follow-up if you want to get this PR merged sooner. I can open a suitable rapids-cmake issue.

Reply:
Opened https://github.com/rapidsai/rapids-cmake/issues/798

Reply:
Tried to describe what this would look like in the context of `rapids-cmake`: https://github.com/rapidsai/rapids-cmake/issues/798#issuecomment-2760017833

Though am sure it can be improved further. Please feel free to suggest there